Add the release note for 1.13.3 (#3943)
line/armeria: site/src/pages/release-notes/1.13.3.mdx

---
date: 2021-11-23
---

## 🌟 New features

- N/A

## 📈 Improvements

- N/A

## 🛠️ Bug fixes

- <type://DecodingClient> does not raise an exception anymore when duplicated encodings are specified. #3932 #3934
  - The duplicated encodings are silently ignored.

> - `unfinishedResponse` is correctly decremented when the <type://HttpResponse> is cancelled. #3920

Reviewer: We can remove this because this was produced by the side effect of #3908 that was not published.

Reviewer: It was my mistake. Please ignore this comment. 😅

> - `unfinishedResponse` is correctly decremented when the <type://HttpResponse> is cancelled. #3920

Reviewer: `unfinishedResponse` is an internal implementation term. Uses might not correctly understand the meaning. 
How about:
```
An early cancelled <type://HttpResponse> is now correctly removed from the active responses.
```

> - The text in the description sections are now correctly rendered in DocService. #3917

Reviewer: ```suggestion
- The text in the description sections is now correctly rendered in <type://DocService>. #3917
```

- The status of a protocol violation is now correctly logged by <type://LoggingService> #3914

>     - A response headers is exactly written once when a protocol violation error is raised.

Reviewer 2: ```suggestion
    - Response headers are written exactly once when a protocol violation error is raised.
```
or
```suggestion
    - A response header is exactly written once when a protocol violation error is raised.
```

Author: We use response headers as a singular form. 😉 
Let me use `A <type://ResponseHeaders>` to clarify that. 😄

> - `GrpcDocStringExtractor` now reads description from `bin`, `desc`, `dsc`, `pb`, and `protobin` file. #3912

Reviewer: ```suggestion
- <type://GrpcDocServicePlugin> now reads description from `bin`, `desc`, `dsc`, `pb`, and `protobin` file. #3912
```

Author: `GrpcDocServicePlugin` is in an internal package so let me use 
`<type://DocServicePlugin> for gRPC` 😉

Reviewer: > GrpcDocServicePlugin is in an internal package

Oops... 😅

- You no longer see `Maximum active streams violated for this endpoint.` exception anymore. #3858 #3908
- <type://EurekaEndpointGroupBuilder> now fails if an empty string is specified for `appName` and
  other parameters. #3903
- OkHttp can be used with self-signed certificate in Armeria. #3902
  - Users can use `https://127.0.0.1:` with the default self-signed certificate.

> - <type://GraphqlService> now produces `graphql+json` response when the `accept` header is `*`. #3898

Reviewer 2: ```suggestion
- <type://GraphqlService> now produces a `graphql+json` response when the `Accept` header is `*`. #3898
```
since `response` is singular

## 🏚️ Deprecations

- N/A

## ☢️ Breaking changes

- N/A

## ⛓ Dependencies

- N/A

## 🙇 Thank you

<ThankYou usernames={[
  'anuraaga',
  'eisig',
  'heowc',
  'hyangtack',
  'ikhoon',
  'jrhee17',
  'minwoox',

>   'ohadgur',

Reviewer 2: can you also add techno who initially reported #3858?

Author: Oops, I didn't notice that. Thanks! 😉

Reviewer: That is a hole in the release note generation script. 🙈

Author: Yeah, I should've double-checked. 😆 
But the script is really great because it reduces the time to create the release note.
